text/000-merge-with-rubygems.md

```diff
@@ -0,0 +1,100 @@
+
+We've been planning and working towards those changes for a long time, and we think you'll find Bundler easier and more straightforward to use once you've had a chance to get used to the changes.
+
+We've already started working on the next version of Bundler, which will be fully part of RubyGems. It will not need to be installed separately, and will always be available anywhere you have Ruby and RubyGems available.
+
+To work towards a combined Bundler and RubyGems, we are combining the Bundler and RubyGems maintainer teams, and we plan to combine GitHub organizations, security teams, Slacks, and eventually repositories and websites.
+
+We're excited about the possibilities opened up by combining the Bundler and RubyGems code, and looking forward to a codebase that is easier to debug, release, and maintain in the future.
+
+**After the code-level merge:**
+
+As of RubyGems 4, Bundler and RubyGems are now merged! That means that they have the same version number, and upgrading RubyGems is how you update to the latest version of Bundler.
+
+As part of this transition, Bundler 4 (as part of RubyGems 4) will be able to be able to install any existing Bundler project: full backwards compatibility.
+
+To use the new, merged Bundler and RubyGems, install the latest RubyGems by running `gem update --system`. You're done! You now have the latest Bundler and the latest RubyGems.
+
+From time to time, Bundler may introduce opt-in changes that require a certain minimum version of Bundler. In that case, it is possible to choose to upgrade an application to require the latest version of Bundler in order to use those features.
+
+However, Bundler will never force an application (or other users of an application) to upgrade to the latest Bundler version. The minimum required version of Bundler will always be opt-in, per application, and never be systemwide across an entire Ruby.
+
+## Reference-level explanation
+
+Implementing the merger will likely require multiple phases. The organizational merger is easy to start right away, while the code-level merger will require some research and experimentation beforehand.
+
+### Phase 1, organization merger (April 2019–May 2019)
+- Move GitHub repos from the `bundler/` org to the `rubygems/` org
+- Add all members of the Bundler maintainers team to the existing RubyGems maintainers team to create a new combined maintainers team
+- Combine maintainer Slack workspaces and Slack channels
+- Combine security@, team@, and other emails
+- Combine READMEs, issue reporting guides, and other in-repo docs
+- Update documentation and website links to issues, pull requests, troubleshooting, developer docs, etc
+
+### Phase 2, research for codebase merger (June 2019–??? 2019)
+- Research and test a merge plan that replaces the Bundler submodule in RubyGems with the entire git history and all commits from the Bundler git repo
```

Comment:
I think I wrote a test script that would do this, I can try and did it up later

Reply:
How will we handle the different code styles / practices used by the two parts of the repo?

Reply:
https://gist.github.com/segiddins/7a846516bd15f6655120db5f91e11e42, found it, but if i'm reading it correctly the point of my script was to make it easy for people with pending contributions to move their PRs over to the rubygems repo

Reply:
A script to help people move their pending PRs over sounds helpful, thanks!